### PR TITLE
Extract crud service

### PR DESCRIPTION
**Pull request by eefscheef**
extract services into ReadService, UpdateService, and CreateService.
these need a custom ReadConverter (with entity to DTO function), UpdateConverter (with DTO to entity update function) and CreateConverter (with entity from DTO function) 
As long as you provide these conversion classes yourself, you gain basic create/read/update functionality (delete is included in ReadService)>

Important change: /invite{id} endpoint was previously checking for Longs in the ID path  param, this has been changed to UUIDs
Also, all nullable ID fields in readDTOs have been non-nulled

### CODE REVIEW

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantReadConverter.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import ut.isep.management.service.converter.ReadConverter
+
+
+@Component
+class ApplicantReadConverter : ReadConverter<Applicant, ApplicantReadDTO> {
+    override fun toDTO(entity: Applicant): ApplicantReadDTO {
+        return ApplicantReadDTO(
+            id = entity.id,
+            name = entity.name,
+            status = entity.status,
```
Comment (SweersJ) [inline management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantReadConverter.kt:16]: is the email missing here?

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantCreateConverter.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import ut.isep.management.service.converter.CreateConverter
+
+@Component
+class ApplicantCreateConverter : CreateConverter<Applicant, ApplicantCreateDTO> {
+
+    override fun fromDTO(createDTO: ApplicantCreateDTO): Applicant {
+        return Applicant(
+            id = 0, // Or some default
+            name = createDTO.name,
+            status = createDTO.status,
```
Comment (SweersJ) [inline management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantCreateConverter.kt:16]: Should this also not have email as an attribute?

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Component
+import ut.isep.management.model.entity.Applicant
+import ut.isep.management.service.converter.UpdateConverter
+
+@Component
+class ApplicantUpdateConverter : UpdateConverter<Applicant, ApplicantUpdateDTO> {
+    override fun updateEntity(entity: Applicant, updateDTO: ApplicantUpdateDTO): Applicant {
+        return entity.apply {
+            updateDTO.name?.let { this.name = it }
+            updateDTO.status?.let { this.status = it }
```
Comment (SweersJ) [inline management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt:14]: same here with email

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            updateDTO.preferredLanguage?.let { this.preferredLanguage = it }
+            updateDTO.score?.let { this.score = it }
```
Comment (SweersJ) [inline management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt:16]: Can you update the score via the applicant Api endpoint? Wouldn't this not be better that system determines this by it self and when you approve a certain answer when you check the assessment results that it automatically updates the score. So then you shouldn't be able to update the score via the applicant Api directly

File: management-server/src/main/kotlin/ut/isep/management/controller/InviteController.kt [hunk 5]
```diff
@@ -77,7 +79,8 @@ class InviteController(val inviteService: InviteService) {
         @RequestBody inviteRequest: InviteCreateDTO
     ): ResponseEntity<URI> {
         return try {
-            val inviteUrl = inviteService.createInvite(inviteRequest)
+            val createdInvite = inviteCreateService.create(inviteRequest)
+            val inviteUrl = URI("https://localhost:8081/invite/${createdInvite.id}")
```
Comment (SweersJ) [inline management-server/src/main/kotlin/ut/isep/management/controller/InviteController.kt]: Can't you do the base path (https://localhost:8081) in a .env and .env.development such that you don't have to change it between development and production

Comment (eefscheef) [reply management-server/src/main/kotlin/ut/isep/management/controller/InviteController.kt]: Addressed in https://github.com/Alakind/ISEP/commit/59086e31403e901350d9403d31b8c04d073e22d1

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt (hunk 4 above)
Comment (eefscheef) [reply management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt:16]: Addressed in https://github.com/Alakind/ISEP/commit/d22ad8043225e091f27c7ecd001049b745356c04

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt (hunk 3 above)
Comment (eefscheef) [reply management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantUpdateConverter.kt:14]: Addressed in https://github.com/Alakind/ISEP/commit/51a8267b109ae3656dd1994b8a210bab7395b205

File: management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantCreateConverter.kt (hunk 2 above)
Comment (eefscheef) [reply management-server/src/main/kotlin/ut/isep/management/service/converter/applicant/ApplicantCreateConverter.kt:16]: Addressed in https://github.com/Alakind/ISEP/commit/1f329faaf32e1020047d9125ff76fcad7bc8190e

**Review by SweersJ — APPROVED** [review]
With one comment that is maybe not yet implemented it is approved

**Pull request merged**